### PR TITLE
[doc] Config and doc updates ahead of KubeRay 0.3.0/Ray 2.0.0

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR
- Updates some configs to refer to Ray 2.0.0
- Updates autoscaler docs to reflect beta status for autoscaler integration
- Makes a couple of grammar edits

It would be good to pick this into the 0.3 branch, since some of the Ray docs refer to the configs in the release branch.

One viable strategy to manage docs for the release branch is just to check the master branch docs straight into the 0.3 branch. 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/guidance/autoscaler.md [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
-To deploy the nightly RayCluster CRD and KubeRay, run
-
+Ray Autoscaler integration is Beta with KubeRay 0.3.0 and Ray 2.0.0.
+Start by deploying the latest stable version of the KubeRay operator:
 ```
-git clone https://github.com/ray-project/kuberay.git
-cd kuberay
-kubectl create -k manifests/cluster-scope-resources
-kubectl apply -k manifests/overlays/autoscaling
+kubectl create -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources?ref=v0.3.0-rc.1&timeout=90s"
```
Comment (DmitriGekhtman) [inline docs/guidance/autoscaler.md:9]: should drop the rc.1 if/when this is picked into the release branch

File: docs/guidance/autoscaler.md [hunk 2]
```diff
@@ -63,21 +62,22 @@ Demands:
...
     ```
 
-2. The autoscaler image is `rayproject/ray:a304d1` which reflects the latest changes from the Ray master branch.
+2. If your RayCluster CR's `spec.rayVersion` field is at least `2.0.0`, the autoscaler container will use the same image as the Ray container.
+   For Ray versions older than 2.0.0, the image `rayproject/ray:2.0.0` will be used to run the autoscaler.
 
-3. Autoscaling functionality is supported only with Ray versions at least as new as 1.11.0. The autoscaler image used
-is compatible with all Ray versions >= 1.11.0.
+3. Autoscaling functionality is supported only with Ray versions at least as new as 1.11.0. Autoscaler support
+   is Beta as of Ray 2.0.0 and KubeRay 0.3.0
```
Comment (edoakes) [inline docs/guidance/autoscaler.md:80]:
```suggestion
   is beta as of Ray 2.0.0 and KubeRay 0.3.0
```
Do we have a link to what this means somewhere? Else just give a short description ("some APIs might change but should be mostly stable" or similar)

**Review by edoakes — APPROVED** [review]
Nice, thanks for driveby grammar fixes.

File: docs/guidance/autoscaler.md (hunk 2 above)
Comment (DmitriGekhtman) [reply docs/guidance/autoscaler.md]: Will add some words.

Comment (DmitriGekhtman) [reply docs/guidance/autoscaler.md]: Words added, thanks.

**Comment by Jeffwan** [comment]
@DmitriGekhtman Do you think if we need to cherry-pick this change to release-0.3 branch?

**Comment by DmitriGekhtman** [comment]
> @DmitriGekhtman Do you think if we need to cherry-pick this change to release-0.3 branch?

We do need to cherry-pick the config changes since we reference the release branch in the Ray docs.
I'll open a PR against the release branch with just the config changes.
 
Other doc changes are kind of optional since we don't do versioned documentation yet.

**Comment by DmitriGekhtman** [comment]
For 0.3 picking, just the sample config changes would be good:
https://github.com/ray-project/kuberay/pull/487

**Pull request merged**